File: 50주차/프린터큐.js
const solution = (idxs, arrs) => {
  arrs.forEach((arr, i) => {
    arr = arr.map((v, i) => [i, v]);
    let nowIdx = idxs[i];

    let cnt = 1;
    while (arr.length) {
      const [idx, num] = arr.shift();
      let flag = false;

      for (let [i, tmp] of arr) {
        if (num < tmp) {
          arr.push([idx, num]);
          flag = true;
          break;
        }
      }
      if (!flag) {
        if (idx === nowIdx) {
          console.log(cnt);
          break;
        }
        cnt += 1;
      }
    }
  });
};
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let tc = null;
let count = 0;
let [n, nowIdx] = [null, null];
let idxs = [];
let arr = [];
rl.on("line", (line) => {
  if (!tc) {
    tc = +line;
  } else if (!n) {
    [n, nowIdx] = line.split(" ").map((e) => +e);
    idxs.push(nowIdx);
  } else {
    arr.push(line.split(" ").map((el) => +el));
    count += 1;

    n = null;
  }
  if (count === tc) {
    rl.close();
  }
}).on("close", () => {
  solution(idxs, arr);
  process.exit();
});
// 백준 문제. 우선순위가 높은것 먼저.처리

File: 52주차/프린터큐.py
from collections import deque
T = int(input())

for _ in range(T):
    n, targetIdx = map(int,input().split())
    cnt  =1
    arr = list(map(int,input().split()))
    q = deque()
    for i,v in enumerate(arr):
        q.append([v,i])
    while q:
        now, idx =q.popleft()
        for v in q:
            if now < v[0]:# 하나라도 큰 값이 있다면, 다시 큐 뒤에 추가.
                q.append([now,idx])
                break
        else:# 한번도 큰 값이 없었다면, 큐에서 빠져나옴. 그 때의 idx가 찾으려는 순서라면 print하기.
            if idx == targetIdx:
                print(cnt)
                break
            cnt += 1
# 백준 큐 자료구조 문제